### DISCUSSION TITLE
Eager loading not working on the `CuratorColumn`

### DISCUSSION THREAD

**Discussion by iRaziul** [awcodes/filament-curator#392] [Q&A]
Most of the time I use `Model::shouldBeStrict()` in my projects.
Filament by default eager loads relationships in the table but I see the error when I use `CuratorColumn` 😟
![image](https://github.com/awcodes/filament-curator/assets/51883557/9d9ecd20-0aaf-4563-afe5-b1aad3b737c3)

At the moment I have to use `getTableQuery` in the list class and load the relationship like this:
![image](https://github.com/awcodes/filament-curator/assets/51883557/84818dd7-1150-41c3-a278-c6e30d7db2dd)

But I think eager loading by default would be a great addition.

**Comment by awcodes** [comment]
Can you explain further. I'm only getting 5 queries, both with and without Model::shouldBeStrict()

![Screenshot 2023-12-22 at 12 13 51 PM](https://github.com/awcodes/filament-curator/assets/3596800/8f994f27-219b-407c-bae3-6024bfdabbe6)

**Comment by iRaziul** [comment]
I think you are using this:
![image](https://github.com/awcodes/filament-curator/assets/51883557/e65c2fee-be60-475b-9f1c-33fcadc34fae)
Here we are already loading the relationship.

Can you try without overriding the `getTableQuery` method?

**Reply by awcodes** [reply]
you're right, sorry. 

I'll see what I can do. Thanks.

**Reply by iRaziul** [reply]
Thanks, I tried to find the solution but couldn't find it due to my lack of `filament core` knowledge 😟

**Comment by awcodes** [comment]
This has been added to the v3.4.0 release. cheers.

**Reply by iRaziul** [reply]
I am facing the same issue after the most recent update 😔

**Reply by Askancy** [reply]
I am also getting the same error after the latest update @awcodes

**Reply by iRaziul** [reply]
I think there will be a major change in the next version where the issue will be fixed.

**Comment by dovaldev** [comment]
I'm not sure if it's the same issue, but I was creating the image column with:

`CuratorColumn::make('logo')->size(40),`

And I solved it by directly calling the relationship.

`CuratorColumn::make('featuredImage')->size(40),`

I have this relationship in the model:

```
public function featuredImage()
{
return $this->belongsTo(Media::class, 'logo', 'id');
}
```

**Reply by iRaziul** [reply]
you are not facing N+1 problem?
